scripts/gb-model/create_region_shapes.py

```
@@ -0,0 +1,421 @@
+            powerplants_gdf = powerplants_gdf.to_crs(regions_gdf.crs)
+            logger.info(f"Converted powerplants CRS from EPSG:4326 to {regions_gdf.crs}")
+        
+        # Find regions that contain powerplants
+        regions_with_powerplants = []
+        powerplant_count_per_region = []
+        
+        for idx, region in regions_gdf.iterrows():
+            # Check which powerplants fall within this region
+            powerplants_in_region = powerplants_gdf[powerplants_gdf.geometry.within(region.geometry)]
+            
+            if len(powerplants_in_region) > 0:
+                regions_with_powerplants.append(region)
+                powerplant_count_per_region.append(len(powerplants_in_region))
+                logger.info(f"Region {region.get('region_id', idx)} contains {len(powerplants_in_region)} powerplants")
+            else:
+                logger.debug(f"Region {region.get('region_id', idx)} contains no powerplants - removing")
+        
+        if regions_with_powerplants:
+            # Create new GeoDataFrame with only regions containing powerplants
+            filtered_regions = gpd.GeoDataFrame(regions_with_powerplants, crs=regions_gdf.crs)
+            filtered_regions['powerplant_count'] = powerplant_count_per_region
+            filtered_regions = filtered_regions.reset_index(drop=True)
+            
+            logger.info(f"Filtered regions: {len(regions_gdf)} -> {len(filtered_regions)} (kept only regions with powerplants)")
+            logger.info(f"Total powerplants in kept regions: {sum(powerplant_count_per_region)}")
+            
+            return filtered_regions
+        else:
+            logger.warning("No regions contain powerplants! Keeping all regions.")
+            return regions_gdf
+            
+    except Exception as e:
+        logger.error(f"Error filtering regions with powerplants: {e}")
+        logger.warning("Continuing with unfiltered regions")
+        return regions_gdf
```

remove, rather let the workflow fail

---

Thanks for pointing, @FabianHofmann. I will remove the function altogether, because that function is no longer in use.